server/websocket_test.go

```diff
@@ -1442,6 +1442,60 @@ func TestWSCompressNegotiation(t *testing.T) {
 	}
 }
 
+func TestWSSetHeader(t *testing.T) {
+	opts := testWSOptions()
+	opts.Websocket.Headers = map[string]string{
+		"X-Header":         "some-value",
+		"X-Another-Header": "another-value",
+	}
+	s := &Server{opts: opts}
+	s.wsSetHeadersOptions(&opts.Websocket)
+	rw := &testResponseWriter{}
+	req := testWSCreateValidReq()
+	res, err := s.wsUpgrade(rw, req)
+	if res == nil || err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	buf := bufio.NewReader(&rw.conn.wbuf)
+	resp, err := http.ReadResponse(buf, req)
+	if err != nil {
+		t.Fatalf("Error reading request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	// Check that the response is a 101
+	if resp.StatusCode != http.StatusSwitchingProtocols {
+		t.Fatalf("Expected 101, got %v", resp.StatusCode)
+	}
+
+	headers := resp.Header.Clone()
+
+	// Compare all the headers
+	for k, v := range opts.Websocket.Headers {
+		if got := headers.Get(k); got != v {
+			t.Fatalf("Expected %q for header %q, got %q", v, k, got)
+		}
+		headers.Del(k)
+	}
+
+	// Check remain headers
+	for k, v := range map[string]string{
+		"Upgrade":              "websocket",
+		"Connection":           "Upgrade",
+		"Sec-Websocket-Accept": wsAcceptKey(req.Header.Get("Sec-Websocket-Key")),
+	} {
+		if got := headers.Get(k); got != v {
+			t.Fatalf("Expected %q for header %q, got %q", v, k, got)
+		}
+		headers.Del(k)
+	}
+
+	for range headers {
```

Comment:
Since it is a `t.Fatalf()` it would stop at the first key/value found, but a `if len(headers)>0{` would be better I think.

Reply:
Because headers is a http.Header is not possible to do a len(leaders) without casting. But if headers have something, it will print all the headers.
Adding a description and also changing the error message to be generic for multiple headers.